vault/auth.go

```diff
@@ -224,6 +224,12 @@ func (c *Core) loadCredentials() error {
 
 		// Upgrade to table-scoped entries
 		for _, entry := range c.auth.Entries {
+			// The auth backend "aws-ec2" was named "aws" in the master.
+			// This is to support upgrade procedure from "aws" to "aws-ec2".
+			if entry.Path == "aws/" {
```

Comment:
This needs to be `entry.Type`. We don't know where they may have mounted it, as that's user specified.